### ISSUE TITLE
transaction ignored silently due to subtraction in amount expression

### ISSUE THREAD

**Issue by andres-erbsen** [ledger/ledger#1809]
```
browse@ashryn ~ % cat bad.ledger                                              
commodity $
payee landlord
account reserves
account housing

2019-07-15 landlord
  ; August (4..23]
  housing  (((23-4)*$85))
  reserves
browse@ashryn ~ % ledger --pedantic --check-payees --immediate -f bad.ledger r                                                                   
browse@ashryn ~ % ledger --pedantic --check-payees --immediate -f bad.ledger b
browse@ashryn ~ % 
```

**Comment by mbbogh** [comment]
The **-** is interpreted as negative sign. Adding a space after it works fine.

`  housing  ((23- 4)*$85)`

Same is true for using an addition.

`  housing  ((23+-4)*$85)`

Looks like a parser error to me.

**Comment by tbm** [comment]
Confirmed, same issue as #2001 which has some debug logs showing the subtraction is incorrectly calculated as 0.

**Comment by tbm** [comment]
Fixed in commit 4ba80c37c068dd1
